src/pages/Encounters/tabs/EncounterNotesTab.tsx

```diff
@@ -317,6 +317,38 @@ const MobileNav = ({
   </div>
 );
 
+const AutoExpandingTextarea = ({
+  value,
+  onChange,
+  onKeyDown,
+  placeholder,
+}: {
+  value: string;
+  onChange: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
+  onKeyDown: (e: React.KeyboardEvent<HTMLTextAreaElement>) => void;
+  placeholder: string;
+}) => {
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+
+  useEffect(() => {
+    if (textareaRef.current) {
+      textareaRef.current.style.height = "auto";
+      textareaRef.current.style.height = `${textareaRef.current.scrollHeight}px`;
+    }
+  }, [value]);
+
+  return (
+    <Textarea
+      ref={textareaRef}
+      placeholder={placeholder}
+      value={value}
+      onChange={onChange}
+      onKeyDown={onKeyDown}
+      className="min-h-[40px] max-h-[200px] resize-none"
+    />
+  );
+};
```

Comment:
move this component into components/ui, either create a new file or put it in textarea